compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -15541,9 +15542,11 @@ private STNode parseAsArrayTypeDesc(STNode typeDesc, STNode openBracket, STNode
         // In ambiguous scenarios typDesc: T[a] may have parsed as an indexed expression.
         // Therefore make an array-type-desc out of it.
         typeDesc = getTypeDescFromExpr(typeDesc);
-        typeDesc = validateForUsageOfVar(typeDesc);
```

Comment:
any reason for removing this validation?

Reply:
Here it was duplicated. this validation is already done inside `createArrayTypeDesc`